File: VisualStudio/CSharp/w7cpframe/clsParameters.cs
//
// Copyright (c)2016 S C Harrison
// Refer to License.txt for terms and conditions of use.
//
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;
using System.Text.RegularExpressions;

namespace w7cpframe
{
    class clsParameters
    {

        public int njt = 0;        //.. No. of joints ..
        public int nmb = 0;        //.. No. of members ..
        public int nmg = 0;        //.. No. of material groups ..
        public int nsg = 0;        //.. No. of member section groups ..
        public int nrj = 0;        //.. No. of supported reaction joints ..
        public int njl = 0;        //.. No. of loaded joints ..
        public int nml = 0;        //.. No. of loaded members ..
        public int ngl = 0;        //.. No. of gravity load cases .. Self weight
        public int nr = 0;         //.. No. of restraints @ the supports ..
        public int mag = 0;        //.. Magnification Factor for graphics

        public void initialise()
        {
            System.Console.WriteLine("initialise ...");
            njt = 0;
            nmb = 0;

            nrj = 0;
            nmg =0;

            nsg =0;
            njl = 0;

            nml = 0;
            ngl = 0;

            nr = 0;

            mag = 0;
            System.Console.WriteLine("... initialise");
        }

        public string sprint()
        {
            string s = "";

            s = s + njt.ToString().PadLeft(6) + nmb.ToString().PadLeft(6);
            s = s + nrj.ToString().PadLeft(6) + nmg.ToString().PadLeft(6);
            s = s + nsg.ToString().PadLeft(6) + njl.ToString().PadLeft(6);
            s = s + nml.ToString().PadLeft(6) + ngl.ToString().PadLeft(6);
            s = s + mag.ToString().PadLeft(6);

            //s = s + StrLPad(this.nr.toString(), 6) + StrLPad(this.mag.toString(), 6);

            return s;
        }

        public void cprint()
        {
            System.Console.WriteLine(sprint());
        }

        public void fprint(StreamWriter fp)
        {
            fp.WriteLine(sprint());
        }

        public void fgetData(StreamReader fp, bool isIgnore)
        {
            string s;
            string [] dataflds = new string[10]; //(0 To 9);
            char [] delimters = new char [] {' ','\t'};
            int i;

            clsParameters myself = new clsParameters();

            System.Console.WriteLine("clsParameters:fgetData ...");

            s = fp.ReadLine();
            System.Console.WriteLine("Source String: <{0}>",s);
            string regResult;
            string pattern = @"^\s+|\s+$"; //trim trailing spaces
            regResult = Regex.Replace(s,pattern,"" );
            System.Console.WriteLine("Trimmed String: <{0}>", regResult);

            //dataflds.Initialize();
            string pattern2 = @"-?\d+(?:[,.]\d+)?";
            i = 0;
            foreach (Match match in Regex.Matches(regResult, pattern2))
            {
                if (match.Value != "")
                {
                    dataflds[i] = match.Value;
                    System.Console.WriteLine("Match: <{0}>", match.Value);
                    i++;
                }
                else 
                {
                    System.Console.WriteLine("RegExp:???? <{0}>", match.Value);
                }
            }

            //typically ignore as all counters are incremented as data read
            //isIgnore=False only used to test parser.
            if (isIgnore)
            {
                //Clear the control data, and count records as read data from file
                initialise(); // (0);
            }
            else
            { //Testing Reading Data

                njt = int.Parse(dataflds[0]);
                nmb = int.Parse(dataflds[1]);
                nrj = int.Parse(dataflds[2]);
                nmg = int.Parse(dataflds[3]);
                nsg = int.Parse(dataflds[4]);
                njl = int.Parse(dataflds[5]);
                nml = int.Parse(dataflds[6]);
                ngl = int.Parse(dataflds[7]);

            }


            if (dataflds[8] != "")
            {
                mag = int.Parse(dataflds[8]);
            }
            else
            {
                mag = 1;
            }

            System.Console.WriteLine("Dimension & Geometry");
            System.Console.WriteLine("-------------------------------");
            System.Console.WriteLine("Number of Joints       : {0}", njt);
            System.Console.WriteLine("Number of Members      : {0}", nmb);
            System.Console.WriteLine("Number of Supports     : {0}", nrj);

            System.Console.WriteLine("Materials & Sections");
            System.Console.WriteLine("-------------------------------");
            System.Console.WriteLine("Number of Materials    : {0}", nmg);
            System.Console.WriteLine("Number of Sections     : {0}", nsg);

            System.Console.WriteLine("Design Actions");
            System.Console.WriteLine("-------------------------------");
            System.Console.WriteLine("Number of Joint Loads  : {0}", njl);
            System.Console.WriteLine("Number of Member Loads : {0}", nml);
            System.Console.WriteLine("Number of Gravity Loads : {0}", ngl);

            System.Console.WriteLine("Screen Magnifier: {0}", mag);

            System.Console.WriteLine("... clsParameters:fgetData");
        }



    } //Class
} //namespace

File: JScript/clsParameters.js
//
// Copyright (c)2014 S C Harrison
// Refer to License.txt for terms and conditions of use.
//



//Define Class
function clsParameters()
{
  this.njt=0;        //.. No. of joints ..
  this.nmb=0;        //.. No. of members ..
  this.nmg=0;        //.. No. of material groups ..
  this.nsg=0;        //.. No. of member section groups ..
  this.nrj=0;        //.. No. of supported reaction joints ..
  this.njl=0;        //.. No. of loaded joints ..
  this.nml=0;        //.. No. of loaded members ..
  this.ngl=0;        //.. No. of gravity load cases .. Self weight
  
  this.nr=0;         //.. No. of restraints @ the supports ..
  
  this.mag=0;        //.. Magnification Factor for graphics
}


clsParameters.prototype.getInitValue = function(baseIndx)
{
  if (baseIndx == 0) {
    return -1;
  } else if (baseIndx == 1) {
    return 0;
  }
}

clsParameters.prototype.initialise = function(baseIndex)
{
    WScript.Echo( "initialise ...");
    this.njt = this.getInitValue(baseIndex);
    this.nmb = this.getInitValue(baseIndex);
    
    this.nrj = this.getInitValue(baseIndex);
    this.nmg = this.getInitValue(baseIndex);
    
    this.nsg = this.getInitValue(baseIndex);
    this.njl = this.getInitValue(baseIndex);
    
    this.nml = this.getInitValue(baseIndex);
    this.ngl = this.getInitValue(baseIndex);
    
    this.nr = this.getInitValue(baseIndex);
    
    this.mag =0;
    WScript.Echo( "... initialise");
}

clsParameters.prototype.sprint = function()
{
  var s = "";
  
  s = s + StrLPad(this.njt.toString(), 6) + StrLPad(this.nmb.toString(), 6);
  s = s + StrLPad(this.nrj.toString(), 6) + StrLPad(this.nmg.toString(), 6);
  s = s + StrLPad(this.nsg.toString(), 6) + StrLPad(this.njl.toString(), 6);
  s = s + StrLPad(this.nml.toString(), 6) + StrLPad(this.ngl.toString(), 6);
  s = s + StrLPad(this.mag.toString(), 6);
  
  //s = s + StrLPad(this.nr.toString(), 6) + StrLPad(this.mag.toString(), 6);
  
  return s;
}

clsParameters.prototype.cprint = function()
{
  WScript.Echo(this.sprint());
}

clsParameters.prototype.fprint = function(fp)
{
  fp.WriteLine(this.sprint());
}

clsParameters.prototype.fgetData = function(fp, isIgnore)
{
  var s;
  var n;
  var dataflds; //(0 To 9);
  var i,n;

  WScript.Echo( "fgetData ...");
  
   s = fp.ReadLine();
   WScript.Echo( s);

   //dataflds = s.split(" "); //No good returns spaces
   s = s.replace(/^\s+|\s+$/gm,''); //trim trailing spaces
   dataflds = s.match(/\s*(\d*)\s*/gi);
   
   
//   n = dataflds.length;
//   for (i=0;i<n;i++)
//   {
//    WScript.Echo( i + "<" + dataflds[i] + ">" );
//   }

   //Call parseDelimitedString(s, dataflds, n, " ")

  //typically ignore as all counters are incremented as data read
  //isIgnore=False only used to test parser.
  if (isIgnore) {
    //Clear the control data, and count records as read data from file
    this.initialise(0);
    
  } else {

    this.njt = parseInt(dataflds[0]);
    this.nmb = parseInt(dataflds[1]);
    this.nrj = parseInt(dataflds[2]);
    this.nmg = parseInt(dataflds[3]);
    this.nsg = parseInt(dataflds[4]);
    this.njl = parseInt(dataflds[5]);
    this.nml = parseInt(dataflds[6]);
    this.ngl = parseInt(dataflds[7]);

  }
 
  
  if (dataflds[8] != "") {
    this.mag = dataflds[8];
  } else {
    this.mag = 1;
  }
  
  WScript.Echo( "Dimension & Geometry");
  WScript.Echo( "-------------------------------");
  WScript.Echo( "Number of Joints       : " + this.njt);
  WScript.Echo( "Number of Members      : " + this.nmb);
  WScript.Echo( "Number of Supports     : " + this.nrj);
  
  WScript.Echo( "Materials & Sections");
  WScript.Echo( "-------------------------------");
  WScript.Echo( "Number of Materials    : " + this.nmg);
  WScript.Echo( "Number of Sections     : " + this.nsg);
  
  WScript.Echo( "Design Actions");
  WScript.Echo( "-------------------------------");
  WScript.Echo( "Number of Joint Loads  : " + this.njl);
  WScript.Echo( "Number of Member Loads : " + this.nml);
  WScript.Echo( "Number of Gravity Loads : " + this.ngl);
  
  WScript.Echo( "Screen Magnifier: " + this.mag);
 
  WScript.Echo( "... fgetData");
}
